File: src/test/java/com/integreety/yatspec/e2e/integration/testapp/external/ExternalClientWithTargetHeader.java
package com.integreety.yatspec.e2e.integration.testapp.external;

import feign.RequestInterceptor;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.context.annotation.Bean;
import org.springframework.web.bind.annotation.PostMapping;

import static com.integreety.yatspec.e2e.integration.testapp.external.ExternalClientWithTargetHeader.ClientConfig;

@FeignClient(name = "externalClientTargetHeader", url = "http://localhost:${wiremock.server.port}", configuration = ClientConfig.class)
public interface ExternalClientWithTargetHeader {

    @PostMapping("/external-api?message=from_feign")
    void post(String message);

    class ClientConfig {
        @Bean
        public RequestInterceptor headersInterceptor() {
            return template -> template.header("Target-Name", "Downstream");
        }
    }
}

File: integrationPostgresTest/src/test/kotlin/io/lsdconsulting/lsd/distributed/interceptor/integration/testapp/external/ExternalClientWithTargetHeader.kt
package io.lsdconsulting.lsd.distributed.interceptor.integration.testapp.external

import feign.RequestInterceptor
import feign.RequestTemplate
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.context.annotation.Bean
import org.springframework.web.bind.annotation.PostMapping

@FeignClient(
    name = "externalClientTargetHeader",
    url = "http://localhost:\${server.wiremock.port}",
    configuration = [ExternalClientWithTargetHeader.ClientConfig::class]
)
interface ExternalClientWithTargetHeader {
    @PostMapping("/external-api?message=from_feign")
    fun post(message: String?)

    class ClientConfig {
        @Bean
        fun headersInterceptor(): RequestInterceptor {
            return RequestInterceptor { template: RequestTemplate ->
                template
                    .header("Authorization", "Password")
                    .header("Target-Name", "Downstream")
            }
        }
    }
}
